File: ruby/reverse_string.rb
def reverse_string(str)
  new_str = ''
  i = 0
  while i < str.length
    new_str = str[i] + new_str
    i += 1
  end
  new_str
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  puts 

  puts "Expecting: 'ocnarf'"
  puts "=>", reverse_string('franco')

  puts "Expecting: 'nairb-oekin'"
  puts "=>", reverse_string('nikeo-brian')

  puts "Expecting: ''"
  puts "=>", reverse_string('')

  # Don't forget to add your own!
end

# Pseudocode:
  # assign a new variable new_str = ''
  # iterate through provided str and add each letter to new_str in reverse order
  # puts new_str
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  let newStr = "";
  for (let i = str.length -1; i >= 0; i --) {
    newStr += str[i];
  }
  return newStr
}

if (require.main === module) {
  // add your own tests in here

  console.log("Expecting: 'oekin'");
  console.log("=>", reverseString("nikeo"));

  console.log("");

  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Pseudocode:
  // create new variable for new string
  //iterate through each character of original string (starting with the last character) and put it in new string in reverse order
  //return new string

// Why it works:
  //This code works by iterating through each letter of the provided string starting from the last letter (represented by i = str.length - 1). Each letter is added to the newStr and then i is increased by 1 so it is doing the second to last letter and so on....until i =0.
